Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride.hs

```diff
@@ -153,3 +176,108 @@ arrivedAtPickup rideId req = do
   pure Success
   where
     isValidRideStatus status = status == DRide.NEW
+
+otpRideCreate ::
+  ( HasCacheConfig r,
+    EsqDBFlow m r,
+    EsqDBReplicaFlow m r,
+    Redis.HedisFlow m r,
+    HasFlowEnv m r '["nwAddress" ::: BaseUrl],
+    HasFlowEnv m r '["selfUIUrl" ::: BaseUrl],
+    HasDriverPoolConfig r,
+    HasHttpClientOptions r c,
+    EncFlow m r,
+    HasShortDurationRetryCfg r c,
+    CoreMetrics m,
+    HasPrettyLogger m r
+  ) =>
+  DP.Person ->
+  OTPRideReq ->
+  m DriverRideRes
+otpRideCreate driver req = do
+  now <- getCurrentTime
+  bookingId <- runInReplica $ QBooking.findBookingBySpecialZoneOTP req.specialZoneOtpCode now >>= fromMaybeM (BookingNotFoundForSpecialZoneOtp req.specialZoneOtpCode)
+  booking <- runInReplica $ QBooking.findById bookingId >>= fromMaybeM (BookingDoesNotExist bookingId.getId)
```

Comment:
why two queries, above query only can return the booking.

Reply:
First I am trying to validate that if booking exists or not just by asking bookingId from booking table.
once I get bookingId then only I am fetching full booking table that contains join with 3 tables , fromLocation, toLocation, FareParams.
So I am trying to avoid unnecessary join.